docs/architecture/microservices/architect-microservice-container-applications/data-sovereignty-per-microservice.md

```diff
@@ -21,7 +21,9 @@ In the traditional approach, there's a single database shared across all service
 
 A monolithic application with typically a single relational database has two important benefits: [ACID transactions](https://en.wikipedia.org/wiki/ACID) and the SQL language, both working across all the tables and data related to your application. This approach provides a way to easily write a query that combines data from multiple tables.
 
-However, data access becomes much more complex when you move to a microservices architecture. But even when ACID transactions can or should be used within a microservice or Bounded Context, the data owned by each microservice is private to that microservice and can only be accessed via its microservice API. Encapsulating the data ensures that the microservices are loosely coupled and can evolve independently of one another. If multiple services were accessing the same data, schema updates would require coordinated updates to all the services. This would break the microservice lifecycle autonomy. But distributed data structures mean that you can't make a single ACID transaction across microservices. This in turn means you must use eventual consistency when a business process spans multiple microservices. This is much harder to implement than simple SQL joins, because you can't create integrity constraints or use distributed transactions between separate databases, as we'll explain later on. Similarly, many other relational database features aren't available across multiple microservices.
+However, data access becomes much more complex when you move to a microservices architecture. But even when ACID transactions can or should be used within a microservice or Bounded Context, the data owned by each microservice is private to that microservice and can only be accessed via its microservice API. The API can be implemented as a REST API, gRPC, SOAP or anything else, but the key point is that the API encapsulates or hides the implementation details.
```

Comment:
I am not sure if we need to be explicit about the implementation details. We can probably, mention using sync and async comm patterns.
```suggestion
However, data access becomes much more complicated when you move to a microservices architecture. Even when using ACID transactions within a microservice or Bounded Context, it is crucial to consider that the data owned by each microservice is private to that microservice and should only be accessed either synchronously through API endpoints or asynchronously via messaging.
```

Reply:
Well, the origin of the change is precisely because, as stated by the user, he's found people often gets equate the API term with the specific REST API implementation. So the change is to enforce those are just implementations of an API.

However, I think that's very good to include the messaging option as just another API implementation.

Does it sound good?

Reply:
Just updated my suggestion.

Reply:
Great!